packages/contracts/contracts/ZERO/CommunityIssuance.sol

```diff
@@ -13,97 +11,145 @@ import "./CommunityIssuanceStorage.sol";
 
-        totalZEROIssued = latestTotalZEROIssued;
-        emit TotalZEROIssuedUpdated(latestTotalZEROIssued);
+        emit APRSet(_APR);
+    }
 
-        return issuance;
+    /**
+     * @dev setter function to set the price feed.
+     * can only be called by the owner.
+     * @param _priceFeedAddress price feed address.
+     */
+    function setPriceFeed(address _priceFeedAddress) external onlyOwner {
+        checkContract(_priceFeedAddress);
+
+        priceFeed = IPriceFeedSovryn(_priceFeedAddress);
+
+        emit PriceFeedAddressSet(_priceFeedAddress);
     }
 
-    /** Gets 1-f^t    where: f < 1
+    /**
+     * @dev setter function to set reward manager.
+     * can only be called by the owner.
+     * @param _rewardManagerAddress reward manager address.
+     */
+    function setRewardManager(address _rewardManagerAddress) external onlyOwner {
+        require(_rewardManagerAddress != address(0), "Account cannot be zero address");
 
-    f: issuance factor that determines the shape of the curve
-    t:  time passed since last ZERO issuance event  */
-    function _getCumulativeIssuanceFraction() internal view returns (uint256) {
-        // Get the time passed since deployment
-        uint256 timePassedInMinutes = block.timestamp.sub(deploymentTime).div(
-            SECONDS_IN_ONE_MINUTE
-        );
+        rewardManager = _rewardManagerAddress;
 
-        // f^t
-        uint256 power = LiquityMath._decPow(ISSUANCE_FACTOR, timePassedInMinutes);
+        emit RewardManagerAddressSet(_rewardManagerAddress);
+    }
 
-        //  (1 - f^t)
-        uint256 cumulativeIssuanceFraction = (uint256(DECIMAL_PRECISION).sub(power));
-        assert(cumulativeIssuanceFraction <= DECIMAL_PRECISION); // must be in range [0,1]
+    /**
+     * @dev validate the APR value.
+     * the value must be >= 0 <= MAX_BPS (10000)
+     */
+    function _validateAPR(uint256 _APR) private {
+        require(_APR <= MAX_BPS, "APR must be less than 10000");
+    }
 
-        return cumulativeIssuanceFraction;
+
+    function issueSOV(uint256 _totalZUSDDeposits) public returns (uint256) {
+        _requireCallerIsStabilityPool();
+
+        uint256 timePassedSinceLastIssuance = (block.timestamp.sub(lastIssuanceTime));
+        uint256 latestTotalSOVIssued = _ZUSDToSOV(_totalZUSDDeposits.mul(APR).div(MAX_BPS).mul(timePassedSinceLastIssuance).div(365 days));
+        
+        uint256 issuance = latestTotalSOVIssued.sub(totalSOVIssued);
+
+        totalSOVIssued = latestTotalSOVIssued;
+        lastIssuanceTime = block.timestamp;
+        emit TotalSOVIssuedUpdated(latestTotalSOVIssued);
+
+        return issuance;
     }
 
-    function sendZERO(address _account, uint256 _ZEROamount) public override {
+    function sendSOV(address _account, uint256 _SOVamount) public {
         _requireCallerIsStabilityPool();
 
-        bool success = zeroToken.transfer(_account, _ZEROamount);
+        bool success = sovToken.transfer(_account, _SOVamount);
```

Comment:
We are transfer ERC20 token here, not a native token.
Or am i missing something here?

Reply:
in case of the sov token we know that there will be no reentrancy on the transfer function because the contract is non-upgradable. in general i agree that reentrancy protection is always a good idea.